spark/v3.5/spark/src/main/java/org/apache/iceberg/spark/source/SparkCopyOnWriteOperation.java

```diff
@@ -95,13 +98,18 @@ public WriteBuilder newWriteBuilder(LogicalWriteInfo info) {
 
   @Override
   public NamedReference[] requiredMetadataAttributes() {
-    NamedReference file = Expressions.column(MetadataColumns.FILE_PATH.name());
-    NamedReference pos = Expressions.column(MetadataColumns.ROW_POSITION.name());
+    List<NamedReference> metadataAttributes = Lists.newArrayList();
+    metadataAttributes.add(Expressions.column(MetadataColumns.FILE_PATH.name()));
 
-    if (command == DELETE || command == UPDATE) {
-      return new NamedReference[] {file, pos};
-    } else {
-      return new NamedReference[] {file};
+    if (TableUtil.supportsRowLineage(table)) {
+      metadataAttributes.add(Expressions.column(MetadataColumns.ROW_POSITION.name()));
+      metadataAttributes.add(Expressions.column(MetadataColumns.ROW_ID.name()));
+      metadataAttributes.add(
+          Expressions.column(MetadataColumns.LAST_UPDATED_SEQUENCE_NUMBER.name()));
+    } else if (command == DELETE || command == UPDATE) {
+      metadataAttributes.add(Expressions.column(MetadataColumns.ROW_POSITION.name()));
     }
+
+    return metadataAttributes.toArray(NamedReference[]::new);
```

Comment:
Checking if I can simplify this a bit more

Reply:
If it's a delete or update add the row position fist. Then if it's row lineage supported table add those columns. That's easier to parse I think compared to the else if structure now